src/main.cpp

```diff
@@ -20,14 +20,25 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-#include "StreamMainWindow.h"
-
+#include <gtest/gtest.h>
 #include <QtWidgets/QApplication>
 
+#include "StreamMainWindow.h"
+
 int main(int argc, char *argv[])
 {
-    QApplication a(argc, argv);
-    StreamMainWindow w;
-    w.show();
-    return a.exec();
+    if(argc < 2) {
+        QApplication a(argc, argv);
+        StreamMainWindow w;
+        w.show();
+        return a.exec();
+    }
+    else {
```

Comment:
We usually separate the gtest executable and the main executable. i.e. They've 2 different main().

Reply:
(Using ctest and gtest_discover_tests)